core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2179,6 +2179,24 @@ private[spark] object Utils extends Logging {
       .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
   }
 
+  val EMPTY_USER_GROUPS = Set[String]()
+
+  // Returns the groups to which the current user belongs.
+  def getCurrentUserGroups(sparkConf: SparkConf, username: String): Set[String] = {
+    val groupProviderClassName = sparkConf.get("spark.user.groups.mapping", "")
```

Comment:
lets leave ShellBasedGroupsMappingProvider as the default because it won't hurt anything since we now catch and ignore exceptions